src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,233 @@
+                        }
+                        else
+                        {
+                            outputBuffer.Append("}");
+                        }
+
+                        position++;
+                        if (position == length)
+                        {
+                            AppendToOutputBuffer(isEncode, arguments, tokenBuffer, ref outputBuffer);
+                            break;
+                        }
+                    }
+                    else
+                    {
+                        AppendToBuffer(isToken, "}", ref tokenBuffer, ref outputBuffer);
+
+                        AppendToOutputBuffer(isEncode, arguments, tokenBuffer, ref outputBuffer);
+
+                        if (position == length)
+                        {
+                            break;
+                        }
+
+                        isToken = false;
+                        tokenBuffer = null;
+                        isEncode = true;
+                    }
+                }
+                else if (currentCharacter == '{')
+                {
+                    if (position < length && resourceString[position] == '{')  // Treat as escape character for {{
+                    {
+                        if (isToken)
+                        {
+                            tokenBuffer.Append("{{");
+                            position++;
+                        }
+                        else
+                        {
+                            position++;
+                            if (position < length &&
+                                (resourceString[position] > '0' && resourceString[position] < '9'))
+                            {
+                                tokenBuffer = new StringBuilder();
+                                tokenBuffer.Append("{{");
+                                isToken = true;
+                            }
+                            else
+                            {
+                                outputBuffer.Append("{");
+                            }
+                        }
+                    }
+                    else
+                    {
+                        tokenBuffer = new StringBuilder();
+                        tokenBuffer.Append("{");
+                        isToken = true;
+                    }
+                }
+                else
+                {
+                    AppendToBuffer(isToken, currentCharacter.ToString(), ref tokenBuffer, ref outputBuffer);
+                }
+            }
+
+            return outputBuffer.ToString();
+        }
+
+        /// <summary>
+        /// Encodes the argument.
+        /// </summary>
+        /// <param name="argument">The object to encode.</param>
+        /// <returns>The encoded object.</returns>
+        private object EncodeArgument([NotNull] object argument)
+        {
+            Debug.Assert(!(argument is HtmlString));
+
+            return _encoder.HtmlEncode(argument.ToString());
```

Comment:
Should inline `_encoder.HtmlEncode(argument.ToString())` into `AppendToOutputBuffer()` since this method is only called in one place and an `HtmlString` is completely impossible there.